lucene/misc/src/test/org/apache/lucene/misc/store/DirectIODirectoryTest.java

```diff
@@ -18,29 +18,37 @@
 
 import com.carrotsearch.randomizedtesting.LifecycleScope;
 import com.carrotsearch.randomizedtesting.RandomizedTest;
-import org.apache.lucene.store.ByteBuffersDirectory;
-import org.apache.lucene.store.Directory;
-import org.apache.lucene.store.IOContext;
-import org.apache.lucene.store.MergeInfo;
+import org.apache.lucene.store.*;
 import org.apache.lucene.util.LuceneTestCase;
-import org.junit.Rule;
-import org.junit.rules.TestRule;
 
 import java.io.IOException;
-import java.util.EnumSet;
+import java.nio.file.Files;
 
-public class NativeUnixDirectoryTest extends LuceneTestCase {
-  @Rule
-  public static TestRule requiresNative = new NativeLibEnableRule(
-      EnumSet.of(NativeLibEnableRule.OperatingSystem.MAC,
-          NativeLibEnableRule.OperatingSystem.FREE_BSD,
-          NativeLibEnableRule.OperatingSystem.LINUX));
+import static org.apache.lucene.misc.store.DirectIODirectory.DEFAULT_MIN_BYTES_DIRECT;
 
-  public void testLibraryLoaded() throws IOException {
+public class DirectIODirectoryTest extends LuceneTestCase {
```

Comment:
In my opinion, this test should also extend the BaseDirectoryTestCase, because it then does a full test of all I/O functionality, including multi-threaded and corner cases - this is important to make sure that the directory works as expected, also if somebody uses it when not merging (e.g. for searching). The original test did not do this as this relied on a native library.

Nevertheless: I am not fully sure if Windows supports this ExtendedOpenOption. I had no time to test this yet.

Reply:
The native version worked for me. I agree with extending base directory test case - I just showed an example of how to make the test run across platforms, even if it required a native dependency.

Reply:
OK, then WindowsDirectory is obsolete, too. Thanks for testing. Nevertheless, as mentioned before, we should add javadocs, that on unsupported platforms, this directory may throw UOE on opening/closing IndexInputs.

We should also extend this test to use a random directory (FS based) as delegate. It looks strange to me to just use a ByteBuffersDirectory as delegate, as the filestore is totally different.

So: Extend the BaseDirectoryTestcase and then create a random temp dir and then build  FSDirectory.open() as backend and use DirectIO as wrapper on top.

Reply:
> Extend the BaseDirectoryTestcase and then create a random temp dir and then build FSDirectory.open() as backend and use DirectIO as wrapper on top.

Done (hope I got it right?)